Update Azure Dev CLI Docs (#34651)
Azure/azure-sdk-for-java: sdk/identity/azure-identity/TROUBLESHOOTING.md

|Azure Service Fabric|[Configuration](https://learn.microsoft.com/azure/service-fabric/concepts-managed-identity)||
|Azure Virtual Machines and Scale Sets|[Configuration](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/qs-configure-portal-windows-vm)|[Troubleshooting](#azure-virtual-machine-managed-identity)|

> *Note*: `azure-identity` v.1.8.1 has a [known issue](https://github.com/Azure/azure-sdk-for-java/issues/34255) with Azure App Service and Azure Functions, upgrade to the latest version v1.8.2 or above.

Reviewer: ```suggestion
*Note*: Version 1.8.1 of `azure-identity` has a [known issue](https://github.com/Azure/azure-sdk-for-java/issues/34255) with Azure App Service and Azure Functions. Upgrade to version v1.8.2 or above.
```

| Error Message |Description| Mitigation |
|---|---|---|
|Azure Developer CLI not installed|The Azure Developer CLI isn't installed or couldn't be found.|<ul><li>Ensure the Azure Developer CLI is properly installed. Installation instructions can be found [here](https://learn.microsoft.com/azure/developer/azure-developer-cli/install-azd).</li><li>Validate the installation location has been added to the `PATH` environment variable.</li></ul>|

> |Please run 'azd auth login' to set up account|No account is currently logged into the Azure Developer CLI, or the login has expired.|<ul><li>Log into the Azure Developer CLI using the `azd login` command.</li><li>Validate that the Azure Developer CLI can obtain tokens. See [below](#verify-the-azure-developer-cli-can-obtain-tokens) for instructions.</li></ul>|

Reviewer: ```suggestion
|Please run 'azd auth login' to set up account|No account is currently logged into the Azure Developer CLI, or the login has expired.|<ul><li>Log into the Azure Developer CLI using the `azd auth login` command.</li><li>Validate that the Azure Developer CLI can obtain tokens. See [below](#verify-the-azure-developer-cli-can-obtain-tokens) for instructions.</li></ul>|
```
